hircluster.c

```diff
@@ -1417,7 +1417,13 @@ static int updateNodesAndSlotmap(redisClusterContext *cc, dict *nodes) {
         dictRelease(oldnodes);
     }
     if (cc->event_callback != NULL) {
-        cc->event_callback(cc, HIRCLUSTER_EVENT_SLOTMAP_UPDATED);
+        cc->event_callback(cc, HIRCLUSTER_EVENT_SLOTMAP_UPDATED,
+                           cc->event_privdata);
+        if (oldnodes == NULL) {
```

Comment:
ahh.. this is what is failing tests..
Since the client adds the init node via the call `redisClusterSetOptionAddNodes(acc->cc, initnode)`
`cc->nodes` will contains this node at start..

Other option would be to use `cc->route_version` which is 0 at start when no sloptmap updated has been performed (but is 1 here since it was increased at line 1407`)